File: src/main/java/com/Ecostyle/CrimsonEyes/model/Usuario.java
package com.Ecostyle.CrimsonEyes.model;

import jakarta.persistence.Entity;
import jakarta.persistence.ForeignKey;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.OneToOne;
import jakarta.persistence.Table;

@Entity
@Table(name = "usuarios")
public class Usuario {

    @Id
    private String email;
    private String password;
    @OneToOne
    @JoinColumn(name = "persona_rut", referencedColumnName = "rut", foreignKey = @ForeignKey(name = "fk_usuario_persona"))
    private Persona persona;

    public Usuario() {
        this.email = "";
        this.password = "";
    }

    public String getEmail() {
        return email;
    }

    public void setEmail(String email) {
        this.email = email;
    }

    public String getPassword() {
        return password;
    }

    public void setPassword(String password) {
        this.password = password;
    }

    public Persona getPersona() {
        return persona;
    }

    public void setPersona(Persona persona) {
        this.persona = persona;
    }

}

File: app/src/main/java/com/example/crimsoneyes/model/Usuario.kt
package com.example.crimsoneyes.model

import androidx.room.Entity
import androidx.room.PrimaryKey
import com.google.gson.annotations.SerializedName

@Entity(tableName = "usuarios")
data class Usuario(
        @PrimaryKey val email: String,
        @SerializedName("password") val password: String = "",
        @SerializedName("rut") val rut: String = "",
        @SerializedName("nombre") val nombre: String = "",
        @SerializedName("telefono") val telefono: Int = 0,
        val id: Long = 0
)
